Automattic/wp-calypso
# Add/migration ssh credentials screen

Related to #77017 

## Proposed Changes

* Add a screen where SSH/FTP credentials can be entered before starting a migration.
* Do not ask for credentials if we found that the source site already has credentials configured.
* Show an `Upgrade your plan` page if the target site is not suitable for a migration.
* Show a helper menu along with the credentials forms to aid the user in finding the appropriate credentials.

## Testing Instructions



* Navigate to [http://calypso.localhost:3000/setup/import-focused/importerWordpress?siteSlug=${target_site}&from=${source_site}&option=everything](http://calypso.localhost:3000/setup/import-focused/importerWordpress?siteSlug=$%7Btarget_site%7D&from=$%7Bsource_site%7D&option=everything)
* Check with an Atomic site and verify that you see the `You are ready to migrate` page.
<img width="826" alt="Screenshot 2566-05-17 at 20 03 31" src="https://github.com/Automattic/wp-calypso/assets/10244734/8db2c590-8577-40ce-ad7e-dd13d88ab73e">

* Check with a simple site and verify that you see the `Upgrade your plan` page.
<img width="777" alt="Screenshot 2566-05-17 at 20 06 40" src="https://github.com/Automattic/wp-calypso/assets/10244734/45b3d061-6774-44e2-908a-0bcb0f7d8172">

* Use a source site without a backup plan and verify that you can enter the credentials of the source site and proceed to migrate.
<img width="1025" alt="Screenshot 2566-05-17 at 20 04 36" src="https://github.com/Automattic/wp-calypso/assets/10244734/294db233-4e25-4cc1-aa80-f9025331342f">

* Use a source site with a backup plan or for which you previously provided credentials and ensure that the `You are ready to migrate` page does NOT display the credentials option.
<img width="829" alt="Screenshot 2566-05-17 at 20 07 59" src="https://github.com/Automattic/wp-calypso/assets/10244734/32a42151-ad95-4292-90ef-01d6609d2aa7">

## Pre-merge Checklist
